### DISCUSSION TITLE
Don't trigger initial page generation during prefetch?

### DISCUSSION THREAD

**Discussion by Maniload** [vercel/next.js#41361] [Help]
Hey, this is probably very similar to this (already fixed) issue: https://github.com/vercel/next.js/issues/17758

What I want to ask is if we could get the same behavior for initial page builds. In my project I often have pages that use the following `getStaticPaths` function:

```ts
export const getStaticPaths: GetStaticPaths = () => {
  return {
    paths: [],
    fallback: "blocking",
  };
};
```

In my app I then have pages with several `<Link>`s to these pages. My problem is that visiting such a page shortly after I deployed my app triggers a lot of page generations. Because of this my server gets overloaded quite a lot.

One possible solution would be to prerender the pages during the build process, but I was wondering if there is another solution for this?

**Comment by icyJoseph** [comment]
You could disable `prefetch` on the `next/link` elements, though it is very manual work, unless you create a thin wrapper around `next/link` where prefetch  defaults to false. Also, `prefetch` only controls generation when the link is visible in the viewport, on hover the page will be generated.

```js
// pseudo-code haven't tested it :) 
const MyLink = ({prefetch = false, ...props}: LinkProps) => <Link prefetch={prefetch} {...props}/> 
```

**Reply by Maniload** [reply]
I think this was also the recommended fix for the issue I mentioned (https://github.com/vercel/next.js/issues/17758) until it was fixed on Next's end. This also disables prefetching for already generated pages though, so it's not ideal. I wonder if this is the intended behavior of Next.js.

**Reply by icyJoseph** [reply]
Well, on hover pre-fetching is still a thing. 

I guess you could do this programmatically, and set a priority prop on your links, disable prefetch and start an effect with a timeout that based on the priority calls `router.prefetch('path')` https://nextjs.org/docs/api-reference/next/router#routerprefetch - the higher the priority the shorter the timeout to prefetch programmatically. 

> Don't forget to check if the link is visible at all in the viewport, and then start the timeout to prefetch

When JS is disabled, the SPA experience if turned off anyway, so there's nothing lost to programatically triggering the prefetching. 

Sorry no code, just words 😅
